lib/matplotlib/ticker.py

```diff
@@ -2451,7 +2451,17 @@ def nonsingular(self, vmin, vmax):
                 "log-scaled.")
             vmin, vmax = 1, 10
         else:
-            minpos = self.axis.get_minpos()
+            ax = self.axis.axes
+            shared_axises = [
+                other_axis
+                for other_ax in [*ax._shared_axes["x"].get_siblings(ax),
```

Comment:
At least this should be `for share in ax._shared_axes.values() for other_ax in share.get_siblings(ax)` inside the comprehension (not hardcoding the xyz) but perhaps the whole thing should really just be `shared_axises = self.axis._get_shared_axis()` (which you define in #24983) as I guess it's really just sharing that matters, not sharing the locator (which anyways should be the same).

The min(...) below also doesn't need to explicitly include self.axis because shared_axises will include self.axis, I think.

Reply:
OK! I was thinking about it, but didn't fully realize the connection... I'll make this draft and get back once #24983 is in.

Reply:
This is now fixed. I wonder if one should do something similar for the other locators?